### ISSUE TITLE
How to utilize the calibration data to evaluate the matches?

### ISSUE THREAD

**Issue by ZeroSteven618** [vcg-uvic/learned-correspondence-release#33]
Hello, I wonder how to use the ground truth data(the calibration data for each image) to get the evaluation for the match correspondences.
Or how to calculate the real perspective transform between 2 images based on the ground truth data?

Thanks for reply!

**Comment by kmyi** [comment]
Hi. We have an extensive dataset [here](https://vision.uvic.ca/image-matching-challenge/data/). This might be what you are looking for. The training / validation data for this challenge should also provide you with calibrations. 

Note: I am not sure what you mean by real perspective transform, but if you are wondering about metric error w.r.t the actual physical world, our current dataset does not have them.

**Comment by ZeroSteven618** [comment]
Well, thanks for your reply!

Maybe **real perspective transform** is not so proper.  
Change the expression:
I want to know: For example, you are doing keypoint matching for imageA and imageB. 

(1)For a keypoint pt1 in imageA, you can calculate the true match correspondence with pt1 in imageB as **pt2_gt** based on the ground truth data(the camera parameters for the 2 images, but I am not sure whether they are enough to be the ground truth data for matching evaluation)

(2)Then you got another match correspondence for pt1 in imageB as **pt2_self** based on your algorithm. 

(3)Next you compare the difference between the **pt2_gt** and **pt2_self**(the distance from coordinate) and the differences from all keypoint-pairs make up the matching accuracy for the image pair matching. Is it right?

I wonder the process for the (1) step. 
In addition, in (3) maybe there is a **threshold** of the difference. If the difference is higher than the threshold the match will be regarded as a wrong match and the lower ones will be regarded as good or right matches? Is it right? If so, what is the common value of the threshold?

Thank you for your reply!

**Comment by kmyi** [comment]
Hi. The very thing that we try to convince people not to do is to evaluate your algorithm based on matching accuracy. They do  not align exactly with the camera pose accuracy. See our [recent paper](https://arxiv.org/abs/2003.01587) for more details. There's a plot that compares matching scores with camera pose accuracy. In this work, instead, you can see that there's a metric we call mAP in this paper, which we changed to mAA in the more recent one, which directly evaluate camera poses.

**Comment by ZeroSteven618** [comment]
So did you utilize the mAA criterion to do evaluation in this repo?
Well, the paper seems complex for me to diagnose in a short time.
I will read it and ask you some more questions. 
Thanks for reply! @kmyi

**Comment by kmyi** [comment]
Yes. See evaluation code. It is approximate though.

**Comment by ZeroSteven618** [comment]
Excuse me, @kmyi . I have another question.
I ran the Preparing Data part for 3 datasets and the Test part as: 
**python main.py --run_mode=test_simple --res_dir="./" --log_dir="models" --test_log_dir="results"  --data_va="reichstag" --data_te="reichstag**"

The execution process seems well and then I got several txt files recording the accuracy in the results file folder. 
But I can not find the matching display for the images(For example the **imwrite** call in OpenCV). 
How to get the matching display images and the matching points information(such as coordinates) in the image pairs?
Thanks for your reply!
